db/migrate/20230816164936_revoke_access_tokens_for_suspended_users.rb

```diff
@@ -0,0 +1,8 @@
+class RevokeAccessTokensForSuspendedUsers < ActiveRecord::Migration[7.0]
+  def up
+    User.with_status(User::USER_STATUS_SUSPENDED)
+      .find_each(&:revoke_all_authorisations)
```

Comment:
@floehopper has [suggested that we shouldn't revoke tokens that are currently, but incorrectly, in use](https://trello.com/c/sIwETh2I/112-prevent-api-tokens-being-created-for-suspended-api-users#comment-64de4eb5f8d80719364c1520). I've had a look but can't see that we have any way to detect when an API token was last used. Do you know if that's possible?

Reply:
@floehopper - how strongly do you feel that we should check whether any of these tokens are being used before revoking them?

Reply:
> @floehopper - how strongly do you feel that we should check whether any of these tokens are being used before revoking them?

I think at the very least we should be letting 2nd Line know we're doing it. Can we chat about it in the morning?

Reply:
In integration there are 10 non-revoked/non-expired access tokens belonging to 6 suspended API users. I'm much less concerned about the impact of revoking these so we haven't investigated them as thoroughly:

```
users = User.with_status(User::USER_STATUS_SUSPENDED).select { |u| u.authorisations.where(revoked_at: nil).any? }
users.count
=> 6

users.select { |u| u.api_user? }.count
=> 6

users.flat_map { |u| u.authorisations.where(revoked_at: nil) }.count
=> 10

users.flat_map { |u| u.authorisations.where(revoked_at: nil) }.map(&:expired?).tally
=> {false=>10}
```

Reply:
Right, I'll update that method on the User model (to only revoke not-yet-revoked tokens) now.

And I think that's the only thing left to do, then?

Reply:
> And I think that's the only thing left to do, then?

Yep! Thanks, Mike :+1: